includes/options/views/class-amp-analytics-options-serializer.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+class Analytics_Options_Serializer {
+
+	public static function save() {
+
+		$option_name = 'analytics';
+
+		if ( !( empty( $_POST['vendor-type'] ) ||
+		     empty( $_POST['config'] ) ) ) {
+
+			if ( empty( $_POST['id-value'] ) ) {
+				$_POST['id-value'] = md5( $_POST['config'] );
+			}
+
+			$new_analytics_options = array(
+				$_POST['id-value'],
+				$_POST['vendor-type'],
```

Comment:
If validation fails when an invalid type is used, we should definitely check against a list -- maybe the UI should provide a dropdown instead? (Only downside would be keeping this list updated.)

Otherwise we can leave is freeform and just sanitize the value using `sanitize_key`.